### PR TITLE
GEM strip definition change

### PR DESCRIPTION
**Pull request by jshlee**
To make all strip/channel definitions unify with HW it was suggested that we change all definitions for strips and channels to start from 0.

Bug in gem digitizer which was causing low efficiency (https://indico.cern.ch/event/748857/contributions/3097829/attachments/1699357/2736224/180809_GEMSimpleModel_bug_fix.pdf)

@dildick @kpedro88 @watson-ij @slowmoyang

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by dildick** [comment]
@jshlee Please also change the documentation here: https://github.com/cms-sw/cmssw/blob/master/DataFormats/GEMDigi/interface/GEMDigi.h#L25 and https://github.com/cms-sw/cmssw/blob/master/DataFormats/GEMDigi/interface/GEMPadDigi.h#L26. 

What about ME0? Do you plan to change conventions there as well?

**Comment by jshlee** [comment]
@slowmoyang will change the other documentation.

For ME0, we plan to leave as is as it will be absorbed by GEM later on.

**Comment by dildick** [comment]
Can you also change the GEM pad digi definition I pointed to? It makes sense to have both strips and pads start from the same number.

**Comment by dildick** [comment]
@@slowmoyang Thank you.

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade
comparison differences restricted to GEM- and muon-related quantities in Phase1/2 workflows

File: Geometry/GEMGeometry/src/GEMEtaPartition.cc [hunk 1]
```diff
@@ -53,7 +53,7 @@ GEMEtaPartition::nstrips() const
 LocalPoint
 GEMEtaPartition::centreOfStrip(int strip) const
 {
-  float s = static_cast<float>(strip) - 0.5;
+  float s = static_cast<float>(strip) + 0.5;
```
Comment (civanch) [inline Geometry/GEMGeometry/src/GEMEtaPartition.cc]: @jshlee , here and in other places better to use 0.5f avoiding type conversion

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
code-checks

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Pull request merged**